lib/spack/spack/fetch_strategy.py

```diff
@@ -421,6 +430,79 @@ def fetch(self):
         tty.msg('Using cached archive: %s' % path)
 
 
+class PyPiFetchStrategy(URLFetchStrategy):
+    """Fetches packages directly from PyPi using XML-RPC to query the API."""
+
+    enabled = True
+    required_attributes = ['pypi']
+
+    def __init__(self, pypi=None, digest=None, **kwargs):
+        # If pypi or digest are provided in the kwargs, then prefer
+        # those values.
+        self.pypi = kwargs.get('pypi', None)
+        if not self.pypi:
+            self.pypi = pypi
```

Comment:
The first thing I noticed when I looked at Spack code was that it did not use normal Python kwarg declarations.  I asked @tgamblin; he told me there was no good reason for this, and that we can start using normal Python.  I recommend we do that; when you read a program and you come across something unusual, you ask "why".  If there's no good reason for it, we're all better off without that unusual thing in place.